tests/js/spec/views/eventsV2/saveQueryButton.spec.jsx

```diff
@@ -0,0 +1,170 @@
+        isEditing={false}
+        eventView={errorsView}
+      />,
+      TestStubs.routerContext()
+    );
+    const button = wrapper.find('StyledDropdownButton');
+    expect(button.text()).toEqual('Save Search');
+  });
+
+  it('renders a popover for a new query', function() {
+    const wrapper = mount(
+      <EventSaveQueryButton
+        organization={organization}
+        location={location}
+        isEditing={false}
+        eventView={errorsView}
+      />,
+      TestStubs.routerContext()
+    );
+    const button = wrapper.find('StyledDropdownButton');
+    button.simulate('click');
+
+    const input = wrapper.find('SaveQueryContainer input');
+    expect(input).toHaveLength(1);
+
+    const submit = wrapper.find('SaveQueryContainer Button');
+    expect(submit).toHaveLength(1);
+    expect(submit.text()).toEqual('Save');
+  });
+
+  it('renders a popover for an existing query', function() {
+    const wrapper = mount(
+      <EventSaveQueryButton
+        organization={organization}
+        location={location}
+        eventView={errorsView}
+        isEditing
+      />,
+      TestStubs.routerContext()
+    );
+    const button = wrapper.find('StyledDropdownButton');
+    button.simulate('click');
+
+    const submit = wrapper.find('SaveQueryContainer Button');
+    expect(submit).toHaveLength(1);
+    expect(submit.text()).toEqual('Update');
+  });
+
+  it('saves a new query', async function() {
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/discover/saved/',
+      method: 'POST',
+      body: {
+        id: '2',
+        name: 'my query',
+        fields: ['title', 'count()'],
+        fieldnames: ['title', 'total'],
+      },
+    });
+    const wrapper = mount(
+      <EventSaveQueryButton
+        organization={organization}
+        location={location}
+        eventView={errorsView}
+      />,
+      TestStubs.routerContext()
+    );
+    const button = wrapper.find('StyledDropdownButton');
+    button.simulate('click');
+
+    const input = wrapper.find('SaveQueryContainer input');
+    input.simulate('change', {target: {value: 'my query'}});
+
+    const submit = wrapper.find('SaveQueryContainer Button');
+    submit.simulate('click');
+
+    // Wait for reflux
+    await tick();
+    await tick();
+
+    // should redirect to query
+    expect(browserHistory.push).toHaveBeenCalledWith({
+      pathname: location.pathname,
+      query: {
+        field: ['title', 'count()'],
+        id: '2',
+        alias: ['title', 'total'],
+        name: 'my query',
+        query: '',
+        sort: [],
+        tag: [],
+      },
+    });
+  });
+
+  it('updates an existing query', async function() {
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/discover/saved/1/',
+      method: 'PUT',
+      body: {
+        id: '1',
+        name: 'my query',
+        fields: ['title', 'count()'],
+        fieldnames: ['title', 'total'],
+      },
+    });
+    errorsView.id = 1;
```

Comment:
Will this lead to unexpected behaviors if we were to add a new test after this (or if tests run out of order)?